prov/efa/src/rxr/rxr_init.c

```diff
@@ -184,14 +183,17 @@ static void rxr_init_env(void)
  *    fe80::4a5:28ff:fe98:e500_0001_12918366_03e8
  *
  * @param[in]	ptr		pointer to raw address (struct efa_ep_addr)
- * @param[out]	smr_name	an unique name for shm ep
- * @return	0 on success.
+ * @param[out]	smr_name	pointer to a char *, which is an unique name for shm ep
+ * @return	0 on success, and smr_name will set to point to an allocated string
+ *              (caller is responsibe to deallocate the memory)
  * 		negative error code on failure.
  */
-int rxr_raw_addr_to_smr_name(void *ptr, char *smr_name)
+int rxr_raw_addr_to_smr_name(void *ptr, char **smr_name)
 {
 	struct efa_ep_addr *raw_addr;
 	char gidstr[INET6_ADDRSTRLEN] = { 0 };
+	char SMR_NAME_FORMAT[] = "%s_%04x_%08x_%04x";
+	int SMR_NAME_BUFF_SIZE = INET6_ADDRSTRLEN + 5 + 9 + 5 + 1;
```

Comment:
Can you drop the magic numbers? Let's just malloc a buffer size that we know is large enough, call snprintf, and validate that the final string size is equal/less than 256.

Reply:
Done in new revision.